### PR TITLE
Fix MIPS PIC level and work around an LLVM bug for `mips(el)-linux-gnueabi(hf)`

### PR DESCRIPTION
**Pull request by alexrp**
* Set the correct PIC level for MIPS.
* Work around https://github.com/ziglang/zig/issues/21215 until https://github.com/llvm/llvm-project/pull/106231 trickles down.
* Revert the workaround for https://github.com/ziglang/zig/issues/14904.

This makes `zig build test-modules -fqemu --glibc-runtimes <...> -Dtest-slow-targets -Dtest-target-filter=mips-linux.4.19...6.10.3-gnueabi -Dtest-target-filter=mipsel-linux.4.19...6.10.3-gnueabi` mostly work, save for an `fchmodat` test failure that I still need to investigate.

### CODE REVIEW

**Comment by The-King-of-Toasters** [comment]
Can you share the `fchmodat` failure here (or in another issue)? I wrote that code so I can give my 2¢.

**Comment by alexrp** [comment]
I don't recall the exact failure, but it was one of the `regfile` asserts here:

https://github.com/ziglang/zig/blob/1a178d499537b922ff05c5d0186ed5a00dbb1a9b/lib/std/posix/test.zig#L1273-L1289

File: src/codegen/llvm/bindings.zig [hunk 1]
```diff
@@ -53,10 +53,10 @@ pub const Module = opaque {
...
     pub const setModulePICLevel = ZigLLVMSetModulePICLevel;
-    extern fn ZigLLVMSetModulePICLevel(module: *Module) void;
+    extern fn ZigLLVMSetModulePICLevel(module: *Module, big: bool) void;
 
     pub const setModulePIELevel = ZigLLVMSetModulePIELevel;
-    extern fn ZigLLVMSetModulePIELevel(module: *Module) void;
+    extern fn ZigLLVMSetModulePIELevel(module: *Module, large: bool) void;
 
     pub const setModuleCodeModel = ZigLLVMSetModuleCodeModel;
     extern fn ZigLLVMSetModuleCodeModel(module: *Module, code_model: CodeModel) void;
```
Comment (jacobly0) [inline src/codegen/llvm/bindings.zig:62]: This API usage should not exist, we already have the ability to modify the module freely.

Comment (alexrp) [reply src/codegen/llvm/bindings.zig:62]: As I understand it, similar to float ABI, you can't set a code model on a bitcode module. So for these, we're stuck with the API (or `llc` later).

PIC/PIE are just module flags though, so I can look into emitting the module flags on our side in a follow-up PR if that's ok.

Comment (jacobly0) [reply src/codegen/llvm/bindings.zig:62]: > As I understand it, similar to float ABI, you can't set a code model on a bitcode module.

Correct for float ABI, but not for code model:
```llvm
!7 = !{i32 1, !"Code Model", i32 1}
```

Comment (alexrp) [reply src/codegen/llvm/bindings.zig:62]: Filed https://github.com/ziglang/zig/issues/21238 so this isn't forgotten.

**Comment by alexrp** [comment]
> I don't recall the exact failure, but it was one of the `regfile` asserts here:

@The-King-of-Toasters

```
test-modules
└─ test-std
   └─ run test std-mips-linux.4.19...6.10.3-gnueabihf.2.28-mips32-Debug-libc 2706/2792 passed, 1 failed, 85 skipped
error: 'posix.test.test.fchmodat smoke test' failed: expected 416, found 488
/home/alexrp/Source/zig/lib/std/testing.zig:97:17: 0xcac51b in expectEqualInner__anon_40880 (test)
                return error.TestExpectedEqual;
                ^
/home/alexrp/Source/zig/lib/std/posix/test.zig:1250:5: 0x25e90e7 in expectMode (test)
    try expectEqual(mode, st.mode & 0b111_111_111);
    ^
/home/alexrp/Source/zig/lib/std/posix/test.zig:1274:5: 0x25e973b in test.fchmodat smoke test (test)
    try expectMode(tmp.dir.fd, "regfile", 0o640);
    ^
error: while executing test 'zig.system.darwin.macos.test.detect', the following test command failed:
qemu-mips -L /opt/glibc/mips-linux-gnueabihf /home/alexrp/Source/zig/.zig-cache/o/a5712f20e8c3b4c16897214a501633b6/test --seed=0x55598d5a --cache-dir=/home/alexrp/Source/zig/.zig-cache --listen=-
```

**Pull request merged**